leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderAPIInteractor.java

```diff
@@ -0,0 +1,92 @@
+package com.schibstedspain.leku.geocoder;
+
+import android.location.Address;
+import android.location.Geocoder;
+import com.google.android.gms.maps.model.LatLng;
+import com.schibstedspain.leku.geocoder.api.AddressBuilder;
+import com.schibstedspain.leku.geocoder.api.NetworkClient;
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import org.json.JSONException;
+import rx.Observable;
+
+public class GeocoderAPIInteractor implements GeocoderInteractorInterface {
+
+  private final static int MAX_RESULTS = 5;
+  private static final String QUERY_REQUEST = "https://maps.googleapis.com/maps/api/geocode/json?address=%1$s&key=%2$s";
+  private static final String QUERY_REQUEST_WITH_RECTANGLE
+      = "https://maps.googleapis.com/maps/api/geocode/json?address=%1$s&key=%2$s&bounds=%3$f,%4$f|%5$f,%6$f";
+  private static final String QUERY_LAT_LONG = "https://maps.googleapis.com/maps/api/geocode/json?latlng=%1$f,%2$f&key=%3$s";
+  private String apiKey;
+  private final NetworkClient networkClient;
+  private final AddressBuilder addressBuilder;
+
+  public GeocoderAPIInteractor(NetworkClient networkClient, AddressBuilder addressBuilder) {
+    this.networkClient = networkClient;
+    this.addressBuilder = addressBuilder;
+  }
+
+  public void setApiKey(String apiKey) {
+    this.apiKey = apiKey;
+  }
+
+  @Override
+  public Observable<List<Address>> getFromLocationName(String query) {
+    return Observable.create(subscriber -> {
+      if (apiKey == null) {
+        subscriber.onCompleted();
+        return;
+      }
+      try {
+        String result = networkClient.requestFromLocationName(String.format(Locale.ENGLISH,
```

Comment:
Why do you use the English locale instead of the default one? It's probably that could be used on different locales.